resources/validations/src/ssp.sch

```diff
@@ -9,33 +8,79 @@
-        <sch:report test="true()">I see <sch:value-of select="count($planned)"/> planned<sch:value-of select="if (count($planned)=1) then ' control implementation' else ' control implementations'"/>.</sch:report>
-        <sch:report test="true()">I see <sch:value-of select="count($all)"/> total<sch:value-of select="if (count($all)=1) then ' control implementation' else ' control implementations'"/>.</sch:report>
+        <sch:assert id="invalid-implemented-requirements-count" test="count($all) > 0">There are no control implementations with statuses set.</sch:assert> 
+        <sch:report id="partial-requirements-report" test="true()">There are <sch:value-of select="count($partial)"/> partial<sch:value-of select="if (count($partial)=1) then ' control implementation' else ' control implementations'"/>.</sch:report>
+        <sch:report id="planned-requirements-report" test="true()">There are <sch:value-of select="count($planned)"/> planned<sch:value-of select="if (count($planned)=1) then ' control implementation' else ' control implementations'"/>.</sch:report>
+        <sch:report id="all-requirements-report" test="true()">There are <sch:value-of select="count($all)"/> total<sch:value-of select="if (count($all)=1) then ' control implementation' else ' control implementations'"/>.</sch:report>
     </sch:rule>
     <sch:rule context="/o:system-security-plan/o:control-implementation">
-        <sch:let name="required" value="$low-p/o:profile/o:import/o:include/o:call"/>
+        <sch:let name="required" value="$selected-profile/*//o:control"/>
         <sch:let name="implemented" value="o:implemented-requirement"/>
-        <sch:let name="missing" value="$required[not(@control-id = $implemented/@control-id)]"/>
-        <sch:report test="true()">The following <sch:value-of select="count($required)"/><sch:value-of select="if (count($required)=1) then ' control' else ' controls'"/> are required: <sch:value-of select="$required/@control-id"/></sch:report>
-        <sch:assert test="count($missing) = 0">This SSP has not implemented <sch:value-of select="count($missing)"/><sch:value-of select="if (count($missing)=1) then ' control' else ' controls'"/>: <sch:value-of select="$missing/@control-id"/></sch:assert>
+        <sch:let name="missing" value="$required[not(@id = $implemented/@control-id)]"/>
+        <sch:report id="each-required-control-report" test="true()">The following <sch:value-of select="count($required)"/><sch:value-of select="if (count($required)=1) then ' control' else ' controls'"/> are required: <sch:value-of select="$required/@id"/></sch:report>
+        <sch:assert id="incomplete-implementation-requirements" test="true()">This SSP has not implemented <sch:value-of select="count($missing)"/><sch:value-of select="if (count($missing)=1) then ' control' else ' controls'"/>: <sch:value-of select="$missing/@id"/></sch:assert>
```

Comment:
locally, i had a merge conflict (simple resolution) when coming from master:HEAD, not sure if you want me to push that merge commit or not

Reply:
So I left this test in half-baked state because I need to debug it further. In short, something with the XSpec global-context-item pattern is not working and this test seems to work purely in Schematron, but fails due to missing context info in XSpec. I am waiting for feedback [from ther XSpec Google Group](https://groups.google.com/g/xspec-users/c/dNvDGZfKVz0). You will observe for now in [the Github Action XSpec results](https://github.com/18F/fedramp-automation/pull/4/checks?check_run_id=1394335757) it is 💤 (hidden/pending) and I marked testing around it that was for now to continue this later in #26 with other stuff once I work this out.

Reply:
makes sense